severity vulnerability reported in the package referenced in vs extension

### Describe the bug

> C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Current\Bin\MSBuild.exe /m /v:m /noAutoRsp /p:Configuration=Release /p:Version=0.62.0 /restore C:\github\typespec-azure\core\packages\typespec-vs\Microsoft.TypeSpec.VS.sln
MSBuild version 17.12.6+db5f6012c for .NET Framework
  Determining projects to restore...
C:\github\typespec-azure\core\packages\typespec-vs\src\Microsoft.TypeSpec.VS.csproj : error NU1902: Warning As Error: Package 'MessagePack' 2.5.124 has a known moderate severity vulnerability, https://github.com/advis
ories/GHSA-4qm4-8hg2-g2xm [C:\github\typespec-azure\core\packages\typespec-vs\Microsoft.TypeSpec.VS.sln]
C:\github\typespec-azure\core\packages\typespec-vs\src\Microsoft.TypeSpec.VS.csproj : error NU1903: Warning As Error: Package 'Microsoft.IO.Redist' 6.0.0 has a known high severity vulnerability, https://github.com/adv 
isories/GHSA-hq7w-xv5x-g34j [C:\github\typespec-azure\core\packages\typespec-vs\Microsoft.TypeSpec.VS.sln]
C:\github\typespec-azure\core\packages\typespec-vs\src\Microsoft.TypeSpec.VS.csproj : error NU1903: Warning As Error: Package 'System.Text.Json' 8.0.0 has a known high severity vulnerability, https://github.com/adviso 
ries/GHSA-8g4q-xg66-9fp4 [C:\github\typespec-azure\core\packages\typespec-vs\Microsoft.TypeSpec.VS.sln]
C:\github\typespec-azure\core\packages\typespec-vs\src\Microsoft.TypeSpec.VS.csproj : error NU1903: Warning As Error: Package 'System.Text.Json' 8.0.0 has a known high severity vulnerability, https://github.com/adviso 
ries/GHSA-hh2w-p6rv-4g7w [C:\github\typespec-azure\core\packages\typespec-vs\Microsoft.TypeSpec.VS.sln]
  Failed to restore C:\github\typespec-azure\core\packages\typespec-vs\src\Microsoft.TypeSpec.VS.csproj (in 1.43 sec).
 ELIFECYCLE  Command failed with exit code 1.

### Reproduction

open typespec vs extension

### Checklist

- [x] Follow our [Code of Conduct](https://github.com/microsoft/typespec/blob/main/CODE_OF_CONDUCT.md)
- [x] Check that there isn't already an issue that request the same bug to avoid creating a duplicate.
- [x] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/Microsoft/typespec/discussions).
- [x] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

--- a/.chronus/changes/vs-vulnerabilities-2024-11-12-13-19-40.md
+++ b/.chronus/changes/vs-vulnerabilities-2024-11-12-13-19-40.md
@@ -1,0 +1,7 @@
+---
+changeKind: internal
+packages:
+  - typespec-vs
+---
+
+Upgrade nuget packages to avoid transitive vulnerabilities

--- a/packages/typespec-vs/src/Microsoft.TypeSpec.VS.csproj
+++ b/packages/typespec-vs/src/Microsoft.TypeSpec.VS.csproj
@@ -33,7 +33,7 @@
     <Reference Include="System.ComponentModel.Composition" />
     <!-- The 'MessagePack' is not used directly but referenced by nuget package 'Microsoft.VisualStudio.SDK', 'Microsoft.VisualStudio.Workspace' and 'Microsoft.VisualStudio.VSIntegration' -->
     <!-- Have to add it explicitly here because the version used by 'Microsoft.VisualStudio...' report vulnerabilities now -->
-    <!-- TODO: remove this explicit reference after 'Microsoft.VisualStudio...' releases new version having 'MessagePack' updated to vulnerability free version-->
+    <!-- TODO: [#5368] remove this explicit reference after 'Microsoft.VisualStudio...' releases new version having 'MessagePack' updated to vulnerability free version-->
     <PackageReference Include="MessagePack" Version="2.5.192" />
     <PackageReference Include="Microsoft.NETFramework.ReferenceAssemblies" Version="1.0.2" />
     <Content Include="Icons\logo.png">
@@ -49,13 +49,13 @@
   <ItemGroup>
     <!-- The 'Microsoft.VisualStudio.Composition' is not used directly but referenced by nuget package 'Microsoft.VisualStudio.Workspace' -->
     <!-- Have to add it explicitly here because otherwise 'Microsoft.VisualStudio.Workspace" will report warning NU1603 because its desired version not found and higher version will be used -->
-    <!-- TODO: remove this explicit reference after 'Microsoft.VisualStudio...' releases new version without the warning-->
+    <!-- TODO: [#5368] remove this explicit reference after 'Microsoft.VisualStudio...' releases new version without the warning-->
     <PackageReference Include="Microsoft.VisualStudio.Composition" Version="17.12.18" />
     <!-- Use 17.0.x or latest 16.x if no 17.0.x for compatible API-->
     <PackageReference Include="Microsoft.VisualStudio.SDK" Version="17.12.40392" ExcludeAssets="Runtime" />
     <!-- The 'Microsoft.VisualStudio.Shell.15.0' is not used directly but referenced by nuget package 'Microsoft.VisualStudio.Workspace.VSIntegration' -->
     <!-- Have to add it explicitly here because otherwise 'Microsoft.VisualStudio.Workspace.VSIntegration" will report warning NU1603 because its desired version not found and higher version will be used -->
-    <!-- TODO: remove this explicit reference after 'Microsoft.VisualStudio...' releases new version without the warning-->
+    <!-- TODO: [#5368] remove this explicit reference after 'Microsoft.VisualStudio...' releases new version without the warning-->
     <PackageReference Include="Microsoft.VisualStudio.Shell.15.0" Version="17.12.40392" />
     <PackageReference Include="Microsoft.VisualStudio.Workspace" Version="17.12.19" ExcludeAssets="Runtime" />
     <PackageReference Include="Microsoft.VisualStudio.Workspace.VSIntegration" Version="17.12.19" ExcludeAssets="Runtime" />

--- a/packages/typespec-vs/src/Microsoft.TypeSpec.VS.csproj
+++ b/packages/typespec-vs/src/Microsoft.TypeSpec.VS.csproj
@@ -21,16 +21,18 @@
     <TargetVsixContainer>$(MSBuildThisFileDirectory)..\$(AssemblyName).vsix</TargetVsixContainer>
   </PropertyGroup>
   <ItemGroup Condition="'$(Configuration)' == 'Debug'">
-    <Content Include="$(IntermediateOutputPath)DebugSourceDirectory.txt" Link="DebugSourceDirectory.txt" Visible="False" />
+    <Content Include="$(IntermediateOutputPath)DebugSourceDirectory.txt"
+      Link="DebugSourceDirectory.txt" Visible="False" />
   </ItemGroup>
   <ItemGroup>
     <Content Include="..\LICENSE" Link="LICENSE" />
     <Content Include="*.pkgdef" />
-    <Content Include="../node_modules/typespec-vscode/dist/typespec.tmLanguage" Link="TextMate/typespec.tmLanguage" />
+    <Content Include="../node_modules/typespec-vscode/dist/typespec.tmLanguage"
+      Link="TextMate/typespec.tmLanguage" />
     <Content Update="@(Content)" IncludeInVSIX="true" CopyToOutputDirectory="PreserveNewest" />
     <Reference Include="System.ComponentModel.Composition" />
     <!-- The 'MessagePack' is not used directly but referenced by nuget package 'Microsoft.VisualStudio.SDK', 'Microsoft.VisualStudio.Workspace' and 'Microsoft.VisualStudio.VSIntegration' -->
-    <!-- Have to add it explicitly here because the vesrion used by 'Micrisoft.VisualStudio...' report vulnerabilities now -->
+    <!-- Have to add it explicitly here because the version used by 'Microsoft.VisualStudio...' report vulnerabilities now -->
     <!-- TODO: remove this explicit reference after 'Microsoft.VisualStudio...' releases new version having 'MessagePack' updated to vulnerability free version-->
     <PackageReference Include="MessagePack" Version="2.5.192" />
     <PackageReference Include="Microsoft.NETFramework.ReferenceAssemblies" Version="1.0.2" />
@@ -59,20 +61,26 @@
     <PackageReference Include="NewtonSoft.JSON" Version="13.0.3" ExcludeAssets="Runtime" />
   </ItemGroup>
   <Import Sdk="Microsoft.NET.Sdk" Project="Sdk.targets" />
-  <Import Project="$(VSToolsPath)\VSSDK\Microsoft.VsSDK.targets" Condition="'$(VSToolsPath)' != '' and '$(MSBuildRuntimeType)' != 'Core'" />
+  <Import Project="$(VSToolsPath)\VSSDK\Microsoft.VsSDK.targets"
+    Condition="'$(VSToolsPath)' != '' and '$(MSBuildRuntimeType)' != 'Core'" />
   <Target Name="GetPackageVersionForVsixManifest" Returns="$(Version)" />
   <!-- VS SDK does not currently support building with dotnet build.  -->
-  <Target Name="WarnAboutDotnetBuildAndVsix" Condition="'$(MSBuildRuntimeType)' == 'Core'" AfterTargets="AfterBuild">
-    <Message Importance="High" Text="VSIX packaging skipped: not supported by `dotnet build`, use Visual Studio `msbuild` to package VSIX." />
+  <Target Name="WarnAboutDotnetBuildAndVsix" Condition="'$(MSBuildRuntimeType)' == 'Core'"
+    AfterTargets="AfterBuild">
+    <Message Importance="High"
+      Text="VSIX packaging skipped: not supported by `dotnet build`, use Visual Studio `msbuild` to package VSIX." />
   </Target>
   <!-- Stash debug source directory so deployed debug vsix can use it to find and tsp-server under
   development in source tree. -->
-  <Target Name="WriteDebugSourceDirectory" Condition="'$(Configuration)' == 'Debug'" BeforeTargets="BeforeBuild">
-    <WriteLinesToFile File="$(IntermediateOutputPath)DebugSourceDirectory.txt" Lines="$(MSBuildThisFileDirectory)" WriteOnlyWhenDifferent="true" Overwrite="true" />
+  <Target Name="WriteDebugSourceDirectory" Condition="'$(Configuration)' == 'Debug'"
+    BeforeTargets="BeforeBuild">
+    <WriteLinesToFile File="$(IntermediateOutputPath)DebugSourceDirectory.txt"
+      Lines="$(MSBuildThisFileDirectory)" WriteOnlyWhenDifferent="true" Overwrite="true" />
   </Target>
   <!-- Helpful for debugging deployment issues to be able to see at minimum verbosity if extension
   was deployed -->
-  <Target Name="LogVsixDeployment" BeforeTargets="DeployVsixExtensionFiles" Condition="'$(DeployExtension)' == 'true'">
+  <Target Name="LogVsixDeployment" BeforeTargets="DeployVsixExtensionFiles"
+    Condition="'$(DeployExtension)' == 'true'">
     <Message Importance="High" Text="$(TargetVsixContainer) -&gt; $(VsixDeploymentPath)" />
   </Target>
 </Project>

--- a/packages/typespec-vs/src/Microsoft.TypeSpec.VS.csproj
+++ b/packages/typespec-vs/src/Microsoft.TypeSpec.VS.csproj
@@ -47,8 +47,16 @@
     </Content>
   </ItemGroup>
   <ItemGroup>
+    <!-- The 'Microsoft.VisualStudio.Composition' is not used directly but referenced by nuget package 'Microsoft.VisualStudio.Workspace' -->
+    <!-- Have to add it explicitly here because otherwise 'Microsoft.VisualStudio.Workspace" will report warning NU1603 because its desired version not found and higher version will be used -->
+    <!-- TODO: remove this explicit reference after 'Microsoft.VisualStudio...' releases new version without the warning-->
+    <PackageReference Include="Microsoft.VisualStudio.Composition" Version="17.12.18" />
     <!-- Use 17.0.x or latest 16.x if no 17.0.x for compatible API-->
     <PackageReference Include="Microsoft.VisualStudio.SDK" Version="17.12.40392" ExcludeAssets="Runtime" />
+    <!-- The 'Microsoft.VisualStudio.Shell.15.0' is not used directly but referenced by nuget package 'Microsoft.VisualStudio.Workspace.VSIntegration' -->
+    <!-- Have to add it explicitly here because otherwise 'Microsoft.VisualStudio.Workspace.VSIntegration" will report warning NU1603 because its desired version not found and higher version will be used -->
+    <!-- TODO: remove this explicit reference after 'Microsoft.VisualStudio...' releases new version without the warning-->
+    <PackageReference Include="Microsoft.VisualStudio.Shell.15.0" Version="17.12.40392" />
     <PackageReference Include="Microsoft.VisualStudio.Workspace" Version="17.12.19" ExcludeAssets="Runtime" />
     <PackageReference Include="Microsoft.VisualStudio.Workspace.VSIntegration" Version="17.12.19" ExcludeAssets="Runtime" />
     <PackageReference Include="Microsoft.VSSDK.BuildTools" Version="17.9.3184">

--- a/packages/typespec-vs/src/Microsoft.TypeSpec.VS.csproj
+++ b/packages/typespec-vs/src/Microsoft.TypeSpec.VS.csproj
@@ -7,6 +7,9 @@
     <AppendTargetFrameworkToOutputPath>false</AppendTargetFrameworkToOutputPath>
     <LangVersion>Latest</LangVersion>
     <Nullable>Enable</Nullable>
+    <!-- Set this to 'all' explicitly to get a consistent behavior. It's the default value now in latest VS/.Net/NugetMgr, but maybe not in some pervious version -->
+    <!-- https://learn.microsoft.com/en-us/dotnet/core/compatibility/sdk/9.0/nugetaudit-transitive-packages -->
+    <NuGetAuditMode>all</NuGetAuditMode>
     <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
     <!-- Official build will pass in the real version from package.json, see scripts/build.js -->
     <Version>42.42.42</Version>
@@ -18,16 +21,18 @@
     <TargetVsixContainer>$(MSBuildThisFileDirectory)..\$(AssemblyName).vsix</TargetVsixContainer>
   </PropertyGroup>
   <ItemGroup Condition="'$(Configuration)' == 'Debug'">
-    <Content Include="$(IntermediateOutputPath)DebugSourceDirectory.txt"
-      Link="DebugSourceDirectory.txt" Visible="False" />
+    <Content Include="$(IntermediateOutputPath)DebugSourceDirectory.txt" Link="DebugSourceDirectory.txt" Visible="False" />
   </ItemGroup>
   <ItemGroup>
     <Content Include="..\LICENSE" Link="LICENSE" />
     <Content Include="*.pkgdef" />
-    <Content Include="../node_modules/typespec-vscode/dist/typespec.tmLanguage"
-      Link="TextMate/typespec.tmLanguage" />
+    <Content Include="../node_modules/typespec-vscode/dist/typespec.tmLanguage" Link="TextMate/typespec.tmLanguage" />
     <Content Update="@(Content)" IncludeInVSIX="true" CopyToOutputDirectory="PreserveNewest" />
     <Reference Include="System.ComponentModel.Composition" />
+    <!-- The 'MessagePack' is not used directly but referenced by nuget package 'Microsoft.VisualStudio.SDK', 'Microsoft.VisualStudio.Workspace' and 'Microsoft.VisualStudio.VSIntegration' -->
+    <!-- Have to add it explicitly here because the vesrion used by 'Micrisoft.VisualStudio...' report vulnerabilities now -->
+    <!-- TODO: remove this explicit reference after 'Microsoft.VisualStudio...' releases new version having 'MessagePack' updated to vulnerability free version-->
+    <PackageReference Include="MessagePack" Version="2.5.192" />
     <PackageReference Include="Microsoft.NETFramework.ReferenceAssemblies" Version="1.0.2" />
     <Content Include="Icons\logo.png">
       <CopyToOutputDirectory>Always</CopyToOutputDirectory>
@@ -41,11 +46,9 @@
   </ItemGroup>
   <ItemGroup>
     <!-- Use 17.0.x or latest 16.x if no 17.0.x for compatible API-->
-    <PackageReference Include="Microsoft.VisualStudio.SDK" Version="17.10.40171" ExcludeAssets="Runtime" />
-    <PackageReference Include="Microsoft.VisualStudio.Workspace" Version="16.3.43"
-      ExcludeAssets="Runtime" />
-    <PackageReference Include="Microsoft.VisualStudio.Workspace.VSIntegration" Version="16.3.43"
-      ExcludeAssets="Runtime" />
+    <PackageReference Include="Microsoft.VisualStudio.SDK" Version="17.12.40392" ExcludeAssets="Runtime" />
+    <PackageReference Include="Microsoft.VisualStudio.Workspace" Version="17.12.19" ExcludeAssets="Runtime" />
+    <PackageReference Include="Microsoft.VisualStudio.Workspace.VSIntegration" Version="17.12.19" ExcludeAssets="Runtime" />
     <PackageReference Include="Microsoft.VSSDK.BuildTools" Version="17.9.3184">
       <PrivateAssets>all</PrivateAssets>
       <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
@@ -56,26 +59,20 @@
     <PackageReference Include="NewtonSoft.JSON" Version="13.0.3" ExcludeAssets="Runtime" />
   </ItemGroup>
   <Import Sdk="Microsoft.NET.Sdk" Project="Sdk.targets" />
-  <Import Project="$(VSToolsPath)\VSSDK\Microsoft.VsSDK.targets"
-    Condition="'$(VSToolsPath)' != '' and '$(MSBuildRuntimeType)' != 'Core'" />
+  <Import Project="$(VSToolsPath)\VSSDK\Microsoft.VsSDK.targets" Condition="'$(VSToolsPath)' != '' and '$(MSBuildRuntimeType)' != 'Core'" />
   <Target Name="GetPackageVersionForVsixManifest" Returns="$(Version)" />
   <!-- VS SDK does not currently support building with dotnet build.  -->
-  <Target Name="WarnAboutDotnetBuildAndVsix" Condition="'$(MSBuildRuntimeType)' == 'Core'"
-    AfterTargets="AfterBuild">
-    <Message Importance="High"
-      Text="VSIX packaging skipped: not supported by `dotnet build`, use Visual Studio `msbuild` to package VSIX." />
+  <Target Name="WarnAboutDotnetBuildAndVsix" Condition="'$(MSBuildRuntimeType)' == 'Core'" AfterTargets="AfterBuild">
+    <Message Importance="High" Text="VSIX packaging skipped: not supported by `dotnet build`, use Visual Studio `msbuild` to package VSIX." />
   </Target>
   <!-- Stash debug source directory so deployed debug vsix can use it to find and tsp-server under
   development in source tree. -->
-  <Target Name="WriteDebugSourceDirectory" Condition="'$(Configuration)' == 'Debug'"
-    BeforeTargets="BeforeBuild">
-    <WriteLinesToFile File="$(IntermediateOutputPath)DebugSourceDirectory.txt"
-      Lines="$(MSBuildThisFileDirectory)" WriteOnlyWhenDifferent="true" Overwrite="true" />
+  <Target Name="WriteDebugSourceDirectory" Condition="'$(Configuration)' == 'Debug'" BeforeTargets="BeforeBuild">
+    <WriteLinesToFile File="$(IntermediateOutputPath)DebugSourceDirectory.txt" Lines="$(MSBuildThisFileDirectory)" WriteOnlyWhenDifferent="true" Overwrite="true" />
   </Target>
   <!-- Helpful for debugging deployment issues to be able to see at minimum verbosity if extension
   was deployed -->
-  <Target Name="LogVsixDeployment" BeforeTargets="DeployVsixExtensionFiles"
-    Condition="'$(DeployExtension)' == 'true'">
+  <Target Name="LogVsixDeployment" BeforeTargets="DeployVsixExtensionFiles" Condition="'$(DeployExtension)' == 'true'">
     <Message Importance="High" Text="$(TargetVsixContainer) -&gt; $(VsixDeploymentPath)" />
   </Target>
 </Project>

--- a/packages/typespec-vs/src/Microsoft.TypeSpec.VS.csproj
+++ b/packages/typespec-vs/src/Microsoft.TypeSpec.VS.csproj
@@ -7,7 +7,7 @@
     <AppendTargetFrameworkToOutputPath>false</AppendTargetFrameworkToOutputPath>
     <LangVersion>Latest</LangVersion>
     <Nullable>Enable</Nullable>
-    <!-- Set this to 'all' explicitly to get a consistent behavior. It's the default value now in latest VS/.Net/NugetMgr, but maybe not in some pervious version -->
+    <!-- Set this to 'all' explicitly to get a consistent behavior. It's the default value now in latest VS/.Net/NugetMgr, but maybe not in some previous version -->
     <!-- https://learn.microsoft.com/en-us/dotnet/core/compatibility/sdk/9.0/nugetaudit-transitive-packages -->
     <NuGetAuditMode>all</NuGetAuditMode>
     <TreatWarningsAsErrors>true</TreatWarningsAsErrors>

